cmp/grid/columns/Column.js

```diff
@@ -238,11 +243,22 @@ export class Column {
             };
         }
 
+        // Support enhanced, absValue-aware sorting via GridSorters in GridModel.sortBy[].
         const sortCfg = find(gridModel.sortBy, {colId: ret.colId});
         if (sortCfg) {
             ret.sort = sortCfg.sort;
             ret.sortedAt = gridModel.sortBy.indexOf(sortCfg);
-            ret.comparator = (v1, v2) => sortCfg.comparator(v1, v2);
+
+            // ag-Grid sort impl. sources its primary values from the node's `groupData` property,
+            // which is not what we want when sorting treeColumns.
+            if (this.isTreeColumn) {
+                ret.comparator = (v1, v2, node1, node2) => {
```

Comment:
Could/Should our sort just always use the valueGetter for the column?  Isn't that more appropriate by default?

Reply:
For ag-grid produced aggregate values you need/want it to look at the groupData to get the computed agg.  That is - I think - why it always looks there - `data[field]` won't have that value.

Reply:
Here is the ValueService within ag-grid where the value is sourced:
https://github.com/ag-grid/ag-grid/blob/1f82c57f6c0035a0f801a0ef58f05c85d0b3ade3/packages/ag-grid/src/ts/valueService/valueService.ts#L51

Note that is the link to the v18 version we are using now.  Here is the latest version, where they seem to have updated things for exactly this issue!!!

https://github.com/ag-grid/ag-grid/blob/master/packages/ag-grid-community/src/ts/valueService/valueService.ts#L51

Can I get three hours of my life back please?